src/firefly/main_window.py

```diff
@@ -269,14 +283,7 @@ def customize_ui(self):
         from .application import FireflyApplication
 
         app = FireflyApplication.instance()
-        app.queue_length_changed.connect(self.set_navbar_visibility)
-
-    @QtCore.Slot(int)
-    def set_navbar_visibility(self, queue_length: int):
-        """Determine whether to make the navbar be visible."""
-        log.debug(f"Setting navbar visibility. Queue length: {queue_length}")
-        navbar = self.ui.navbar
-        navbar.setVisible(queue_length > 0)
+        app.queue_in_use_changed.connect(self.ui.navbar.setVisible)
```

Comment:
I have a question: does the navbar only become visible in the Firefly main window? Or it will become also visible in all the plan windows as well?

Reply:
It only becomes visible on ``PlanWindow`` instances. Other instances of ``FireflyMainWindow`` do not show the queue navbar. The reason is that otherwise even small windows, like motor controls, would have this big navbar at the top and that seemed unnecessary. Thoughts?

Reply:
Agree! Ok, actually I wanted to ask, e.g., if we have a LineScan plan running, and I am opening a XafsScan plan window. Does the NavBar shows the LineScan status or will only show the XafsScan status?

Reply:
The navbar has the queueserver actions in it. It doesn't matter where the items in the queue came from. I think that's what you're asking?

Reply:
so what you're saying is the Navbar on the XafsScan plan may control a different scan being queued. I guess that might be a little bit confusing, cause ppl might stop the queue item in the XafsScan window thinking that they are stopping the XafsScan queue items but it is actually stopping the other items running in the queue. Nevertheless I think let's test this version now first. It can be that I overthink about this senario, and I don't want to non-necessary efforts/complexities..